### ISSUE TITLE
Getting Item Taken from a Slot?

### ISSUE THREAD

**Issue by vhorvath2010** [IPVP-MC/canvas#23]
Is there a way to get what item was removed from a slot? I'm currently trying:

https://paste.md-5.net/ofogaqoqah.coffeescript

and I'm getting an error that the itemstack in that slot is null.

**Comment by vhorvath2010** [comment]
Update, I tried this and it also gives a null pointer: https://paste.md-5.net/weyosicude.coffeescript

**Comment by sainttx** [comment]
I'll need to test out a few cases, but it could be worth trying `getClickedSlot().getRawItem()`.

**Comment by vhorvath2010** [comment]
Thanks for the suggestion. I'll try when I'm home and if it works I'll make a PR to make `getItemAmount()` more robust and add a corresponding `getTakingItem()` to the ClickInformation class

**Comment by vhorvath2010** [comment]
> I'll need to test out a few cases, but it could be worth trying `getClickedSlot().getRawItem()`.

This works, I'll make a PR with the suggested changes above

**Comment by vhorvath2010** [comment]
https://github.com/IPVP-MC/canvas/pull/24

**Comment by vhorvath2010** [comment]
@sainttx Asking for a review on this [PR](https://github.com/IPVP-MC/canvas/pull/24) if you'd like to include these changes

**Comment by sainttx** [comment]
Hey, will try to carve out some time this weekend to take a look. My main concern is backwards-compat so will need to test out how the changes impact the API.

**Comment by sainttx** [comment]
Hey @vhorvath2010, sorry for the 2 month delay. I liked your idea for the taken item API but discovered that canvas had some issues with the `SWAP_WITH_CURSOR` InventoryAction (and a few others) that I wanted to address with the new APIs so I've implemented in one of the newer commits.

I've gone ahead and expanded `ClickInformation` with the following methods:
* `getAddingItemAmount` to get the quantity of item added to the clicked slot
* `getTakingItem` to get the item being removed
* `getTakingItemAmount` to get the quantity of item being removed

Additionally, `ClickInformation#getAddingItem` now properly handles hotbar swapping.

I also didn't want to break functionality for existing users that might be using `getItemAmount` so I simply deprecated the method for now.
